src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java

```diff
@@ -0,0 +1,131 @@
+    private final AirGapDownloadStrategy airGapDownloadStrategy;
+
+    public DetectAirGapJarStrategy(JenkinsIntLogger logger, IntEnvironmentVariables intEnvironmentVariables, String remoteJdkHome, JenkinsConfigService jenkinsConfigService, AirGapDownloadStrategy airGapDownloadStrategy) {
+        this.logger = logger;
+        this.intEnvironmentVariables = intEnvironmentVariables;
+        this.remoteJdkHome = remoteJdkHome;
+        this.jenkinsConfigService = jenkinsConfigService;
+        this.airGapDownloadStrategy = airGapDownloadStrategy;
+    }
+
+    @Override
+    public Function<String, String> getArgumentEscaper() {
+        return Function.identity();
+    }
+
+    @Override
+    public MasterToSlaveCallable<ArrayList<String>, IntegrationException> getSetupCallable() {
+        return new SetupCallableImpl(logger, intEnvironmentVariables.getVariables(), remoteJdkHome, jenkinsConfigService, airGapDownloadStrategy);
+    }
+
+    public static class SetupCallableImpl extends MasterToSlaveCallable<ArrayList<String>, IntegrationException> {
+        private static final long serialVersionUID = -8326836838838706367L;
+
+        private final JenkinsIntLogger logger;
+        private final Map<String, String> environmentVariables;
+        private final String remoteJdkHome;
+        private final JenkinsConfigService jenkinsConfigService;
+        private final AirGapDownloadStrategy airGapDownloadStrategy;
+
+        public SetupCallableImpl(JenkinsIntLogger logger, Map<String, String> environmentVariables, String remoteJdkHome, JenkinsConfigService jenkinsConfigService, AirGapDownloadStrategy airGapDownloadStrategy) {
+            this.logger = logger;
+            this.environmentVariables = environmentVariables;
+            this.remoteJdkHome = remoteJdkHome;
+            this.jenkinsConfigService = jenkinsConfigService;
+            this.airGapDownloadStrategy = airGapDownloadStrategy;
+        }
+
+        @Override
+        public ArrayList<String> call() throws IntegrationException {
+            String airGapJarPath = getOrDownloadAirGapJar();
+            RemoteJavaService remoteJavaService = new RemoteJavaService(logger, remoteJdkHome, environmentVariables);
+            String javaExecutablePath = remoteJavaService.calculateJavaExecutablePath();
+            remoteJavaService.logDebugData(javaExecutablePath);
+
+            logger.info("Detect configured: " + airGapJarPath);
+
+            return new ArrayList<>(Arrays.asList(javaExecutablePath, "-jar", airGapJarPath));
+        }
+
+        private String getOrDownloadAirGapJar() throws IntegrationException {
+            DetectAirGapInstallation airGapInstallation;
+            try {
+                airGapInstallation = jenkinsConfigService.getInstallationForNodeAndEnvironment(DetectAirGapInstallation.DescriptorImpl.class, airGapDownloadStrategy.getAirGapInstallationName()).get();
+            } catch (Exception e) {
```

Comment:
We're swallowing an InterruptedException here, which we don't want to do.

See: https://sonarcloud.io/organizations/black-duck-software/rules?open=java%3AS2142&q=InterruptedException